x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -100,20 +104,50 @@ export function trainedModelsRoutes({ router, routeGuard }: RouteInitialization)
                   ...Object.values(modelDeploymentsMap).flat(),
                 ])
               );
+              const modelsClient = modelsProvider(client);
 
-              const pipelinesResponse = await modelsProvider(client).getModelsPipelines(
-                modelIdsAndAliases
+              const modelsPipelinesAndIndices = await Promise.all(
```

Comment:
This API is intended for retrieving the model info. I've extended the ES response with pipelines definitions if `with_pipelines` query param is provided. I see you extended it with `indices`, so we can do the same, i.e. only include them when `with_indices` or `include_indices` is provided.

Reply:
Updated with tests here [`ecc7493` (#162853)](https://github.com/elastic/kibana/pull/162853/commits/ecc74937d95b619907f9e2567fda064add8944cb)